ffi/derive/src/util.rs

```diff
@@ -39,42 +39,41 @@ pub fn gen_derived_methods(item: &syn::ItemStruct) -> Vec<FnDescriptor> {
 }
 
 pub fn gen_arg_ffi_to_src(arg: &Arg, is_output: bool) -> TokenStream {
-    let (arg_name, src_type) = (arg.name(), arg.src_type_resolved());
+    let (arg_name, src_type) = (arg.name(), arg.src_type_resolved(false));
+    let store_name = gen_store_name(arg_name);
 
     if is_output {
         let mut stmt = quote! {
-            let mut store = ();
-            let #arg_name: #src_type = iroha_ffi::TryFromReprC::try_from_repr_c(#arg_name, &mut store)?;
+            let mut #store_name = ();
+            let #arg_name: #src_type = iroha_ffi::FfiType::try_from_ffi(#arg_name, &mut #store_name)?;
         };
 
         if let Type::Reference(ref_type) = &src_type {
             let elem = &ref_type.elem;
 
+            // TODO: Seems that this hack can be removed now?
```

Comment:
Could you provide more context?

Reply:
it's not relevant anymore, I've updated the code with attached comment